graylog2-server/src/main/java/org/graylog2/rest/resources/roles/RolesResource.java

```diff
@@ -155,10 +155,30 @@ public RoleResponse update(
 
     @DELETE
     @Path("{rolename}")
-    @ApiOperation(value = "Remove the named role")
+    @ApiOperation(value = "Remove the named role and unassign any users from it")
     public void delete(@ApiParam(name = "rolename", required = true) @PathParam("rolename") String name) throws NotFoundException {
         checkPermission(RestPermissions.ROLES_DELETE, name);
-        // TODO unassign roles from current role members
+
+        final Role role = roleService.load(name);
+        if (role.isReadOnly()) {
+            throw new BadRequestException("Cannot delete read only system role " + name);
+        }
+        final Collection<User> usersInRole = userService.loadAllForRole(role);
+        // remove role from any user still assigned
+        for (User user : usersInRole) {
```

Comment:
How about moving the user related stuff into the `UserService`? That cleans up the resource and the code might be useful in other places as well.